src/mainwindow.cpp

```diff
@@ -418,6 +493,85 @@ void MainWindow::on_openFileButton_clicked()
     openFile(fileName);
 }
 
+void MainWindow::onRecordDataButtonClicked()
+{
+    ui->mainPageStack->setCurrentWidget(ui->recordDataPage);
+}
+
+void MainWindow::onHomeButtonClicked()
+{
+    ui->mainPageStack->setCurrentWidget(ui->startPage);
+}
+
+void MainWindow::onStartRecordingButtonClicked(bool checked)
+{
+    if (checked) {
+        ui->startRecordingButton->setIcon(QIcon::fromTheme(QStringLiteral("media-playback-stop")));
+        ui->startRecordingButton->setText(tr("Stop Recording"));
+        ui->perfResultsTextEdit->clear();
+
+        QStringList perfOptions;
+        const auto callGraphOption = ui->callGraphComboBox->currentData().toString();
+
+        if (!callGraphOption.isEmpty()) {
+            perfOptions << QStringLiteral("--call-graph") << callGraphOption;
+        }
+
+        if (!ui->eventTypeBox->text().isEmpty()) {
+            perfOptions << QStringLiteral("--event") << ui->eventTypeBox->text();
+        }
+
+        m_perfRecord->record(perfOptions, ui->workingDirectory->text() + QStringLiteral("/perf.data"),
+                             ui->applicationName->text(), KShell::splitArgs(ui->applicationParametersBox->text()));
+    } else {
+        ui->startRecordingButton->setIcon(QIcon::fromTheme(QStringLiteral("media-playback-start")));
+        ui->startRecordingButton->setText(tr("Start Recording"));
+        m_perfRecord->stopRecording();
+    }
+}
+
+void MainWindow::onApplicationNameChanged(const QString& filePath)
+{
+    QFileInfo application(filePath);
+
+    if (!application.exists()) {
+        ui->applicationRecordErrorLabel->setText(tr("Application file cannot be found: ") + filePath);
+    } else if (!application.isFile()) {
+        ui->applicationRecordErrorLabel->setText(tr("Application file is not valid: ") + filePath);
+    } else if (!application.isExecutable()) {
+        ui->applicationRecordErrorLabel->setText(tr("Application file is not executable: ") + filePath);
+    } else {
+        if (ui->workingDirectory->text().isEmpty()) {
+            ui->workingDirectory->setPlaceholderText(QFileInfo(filePath).path());
+        }
+        ui->applicationRecordErrorLabel->hide();
+        return;
+    }
+    ui->applicationRecordErrorLabel->show();
+}
+
+void MainWindow::onWorkingDirectoryNameChanged(const QString& folderPath)
+{
+    QFileInfo folder(folderPath);
+
+    if (!folder.exists()) {
+        ui->applicationRecordErrorLabel->setText(tr("Working directory folder cannot be found: ") + folderPath);
```

Comment:
This might not work in all languages. It's better to use `tr("Foo %1").arg(folderPath)`